@martinwicke Sorry, this python unit test has no problem on my code base. I have debugged the code inside and I am sure the lambda regularization loss has been added into the ops.GraphKeys.REGULARIZATION_LOSSES collection. So What's the version of TF does ci build test use? I have tested on master branch. 

FAIL: testReusePartitionedVaraiblesAndRegularizers (__main__.BaseLayerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kbuilder/.cache/bazel/_bazel_kbuilder/f2d52ca1f092ccbe254cc98c3dc90790/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/python/layers_base_test.runfiles/org_tensorflow/tensorflow/python/layers/base_test.py", line 111, in testReusePartitionedVaraiblesAndRegularizers
    self.assertEqual(len(ops.get_collection(ops.GraphKeys.REGULARIZATION_LOSSES)), 3)
AssertionError: 0 != 3

--- a/tensorflow/python/layers/base.py
+++ b/tensorflow/python/layers/base.py
@@ -197,6 +197,7 @@ class Layer(base_layer.Layer):
         for var in variable:
           if var in existing_variable_set:
             return False
+        return True
       else:
         return variable not in existing_variable_set
 
@@ -240,7 +241,6 @@ class Layer(base_layer.Layer):
             partitioner=partitioner,
             use_resource=use_resource,
             getter=vs.get_variable)
-
         if regularizer:
           if context.executing_eagerly() or _should_add_regularizer(
               variable, existing_variables):

--- a/tensorflow/python/layers/base.py
+++ b/tensorflow/python/layers/base.py
@@ -241,6 +241,7 @@ class Layer(base_layer.Layer):
             partitioner=partitioner,
             use_resource=use_resource,
             getter=vs.get_variable)
+
         if regularizer:
           if context.executing_eagerly() or _should_add_regularizer(
               variable, existing_variables):

--- a/tensorflow/python/layers/base_test.py
+++ b/tensorflow/python/layers/base_test.py
@@ -108,7 +108,8 @@ class BaseLayerTest(test.TestCase):
             'reg_part_var', [4, 4],
             initializer=init_ops.zeros_initializer(),
             regularizer=regularizer)
-    self.assertEqual(len(ops.get_collection(ops.GraphKeys.REGULARIZATION_LOSSES)), 3)
+    self.assertEqual(
+        len(ops.get_collection(ops.GraphKeys.REGULARIZATION_LOSSES)), 3)
 
   def testNoEagerActivityRegularizer(self):
     with context.eager_mode():

